ksql-serde/src/main/java/io/confluent/ksql/serde/json/KsqlJsonDeserializer.java

```diff
@@ -163,34 +164,31 @@ private static Struct enforceFieldTypesForStruct(final JsonValueContext context)
     }
 
     final Struct columnStruct = new Struct(context.schema);
-    final Map<String, ?> fields =
-        toCaseInsensitiveFieldNameMap((Map<String, ?>) context.val, context.topLevel);
-
-    context.schema.fields().forEach(
-        field -> {
-          final Object fieldValue = fields.get(field.name().toUpperCase());
-          final Object coerced = enforceFieldType(
-              context.deserializer, field.schema(), fieldValue, false);
-          columnStruct.put(field.name(), coerced);
-        });
+    final Map<String, ?> fields = (Map<String, ?>) context.val;
+    final Map<String, ?> caseInsensitiveFields = toCaseInsensitiveFieldNameMap(fields);
+
+    for (Field field : context.schema.fields()) {
+      final Object fieldValue = ObjectUtils.defaultIfNull(
+          fields.get(field.name()),
+          caseInsensitiveFields.get(field.name().toUpperCase()));
```

Comment:
We only want to do this for fields that are not quoted, right? Might be good to have some docs that specify how json field names map to KSQL field names.

Reply:
Good point - since the new model is for all KSQL names to be case sensitive, the unquoted ones will be uppercased at parse-time (in AstBuilder). I will remove the `toUpperCase` call on the `ksqlFieldName` and have a comment explaining that.